scripts/globals/pets/avatar.lua

```diff
@@ -0,0 +1,385 @@
+-- print('onMobMagicPrepare: ' .. pet:getMobMod(xi.mobMod.MAGIC_COOL))
+    local master = pet:getMaster()
+
+    if master and master:getObjType() == xi.objType.PC then
+        -- meta checks for fresh pet, etc
+        pet:setLocalVar(lastCastTimeVar, 0)
+        pet:setLocalVar(lastCastTimeStampVar, os.time())
+        local spellID = 0
+        local spellTarget = nil
+        if pet:getMobMod(xi.mobMod.MAGIC_COOL) == 1 then
+            setMagicCastCooldown(pet)
+
+            return -1
+        end
+
+        pet:setLocalVar(buffModeVar, 1)
+
+        -- Core functionality to decide which spell to use
+        spellID, spellTarget = xi.pets.avatar.getSpiritSpell(pet)
+
+        -- Final items to cast the spell and ensure cast delay is proper
+        local spell = GetSpell(spellID)
+        if spell then
+            if spell:getSkillType() == xi.skill.ENHANCING_MAGIC then
+                -- half casting delay
+                pet:setLocalVar(buffModeVar, 0)
+            end
+-- print(spellTarget:getName() .. '->' .. spellID)
+            pet:castSpell(spellID, spellTarget or pet)
+            setMagicCastCooldown(pet)
+            return -1
+        end
+    end
+end
+
+xi.pets.avatar.getSpiritSpell = function(pet)
+    local spellID = 0
+    local spellTarget = nil
+    local petID = pet:getPetID()
+    if petID == xi.petId.LIGHT_SPIRIT then
+        spellID, spellTarget = xi.pets.avatar.getLightSpiritSpell(pet)
+    -- elseif not pet:isEngaged() then
+    --     -- to show how we can further tune idle casting
+    --     if not pet:hasStatusEffect(xi.effect.SHOCK_SPIKES) then
+    --         spellID = xi.magic.spell.SHOCK_SPIKES
+    --         spellTarget = pet
+    --     end
+    end
+
+    return spellID, spellTarget
+end
+
+-- Disable cyclomatic complexity check for this function:
+-- luacheck: ignore 561
+xi.pets.avatar.getLightSpiritSpell = function(pet)
+    -- returns the spirit's preferred target based on positioning
+    local master = pet:getMaster()
+    local posTarget = nil
+    local posSpellID = 0
+    local buffTarget = nil
+    local buffSpellID = 0
+    local cureTarget = nil
+    local spellID = 0
+
+    if master then
+        if not pet:isEngaged() then
+            pet:lookAt(master:getPos())
+        else
+            -- don't do logic to buff anyone if pet is engaged. buffTarget will be reset below
+            buffTarget = master
+        end
+
+        local distance  = pet:checkDistance(master) -- starts as distance to master, updated to be distance to the posTarget
+        local hpp       = 0
+        local alliance  = master:getAlliance()
+        local party     = master:getParty()
+        local petLvl    = pet:getMainLvl()
+
+        for _, member in pairs(alliance) do
+            -- tiered chance of healing based on how low they are
+            local tempHPP = member:getHPP()
+            if
+                pet:checkDistance(member) < 20 and
```

Comment:
This if statement is very condensed and confusing, can you break it out into sub-conditions with relevant names, or into multiple checks wrapped in a table?

Reply:
Cleaned this up to be more obvious how the logic flows. Still works to have a chance of not healing yellow/organge players:

![image](https://github.com/LandSandBoat/server/assets/131182600/99bbeb20-5b18-4677-8416-3e2f3239adcd)